rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -705,3 +709,75 @@ def parse_service_string(pkg_name, srv_name, message_string):
         pkg_name, srv_name + SERVICE_RESPONSE_MESSAGE_SUFFIX, response_message_string)
 
     return ServiceSpecification(pkg_name, srv_name, request_message, response_message)
+
+def parse_action_file(pkg_name, interface_filename):
+    basename = os.path.basename(interface_filename)
+    action_name = os.path.splitext(basename)[0]
+    with open(interface_filename, 'r') as h:
+        return parse_action_string(pkg_name, action_name, h.read())
+
+
+def parse_action_string(pkg_name, action_name, message_string):
+    lines = message_string.splitlines()
+    separator_indices = [
```

Comment:
@sservulo nit: another option would be to split the whole definition into sections separated by "---" lines and then work with those. You wouldn't have to play with slices then.